integration-tests/q-agentic-chat-server/src/tests/lspClient.ts

```diff
@@ -0,0 +1,126 @@
+import { JSONRPCClient } from 'json-rpc-2.0'
+import { EventEmitter } from 'events'
+import { Readable, Writable } from 'stream'
+import {
+    ButtonClickParams,
+    ButtonClickResult,
+    EncryptedChatParams,
+    InitializeParams,
+    InitializeResult,
+} from '@aws/language-server-runtimes/protocol'
+
+/**
+ * JSON-RPC endpoint for communicating with Language Server Protocol (LSP) servers.
+ *
+ * Acts as a JSON-RPC client that sends requests/notifications to a server process
+ * and receives responses/notifications back. Uses LSP-style message framing with
+ * Content-Length headers.
+ *
+ * @example
+ * ```typescript
+ * const endpoint = new JSONRPCEndpoint(process.stdin, process.stdout);
+ *
+ * // Send request and wait for response
+ * const result = await endpoint.send('initialize', { capabilities: {} });
+ *
+ * // Send notification (no response expected)
+ * endpoint.notify('initialized', {});
+ *
+ * // Listen for server notifications
+ * endpoint.on('window/logMessage', (params) => {
+ *   console.log('Server log:', params.message);
+ * });
+ * ```
+ */
+export class JSONRPCEndpoint extends EventEmitter {
+    private client: JSONRPCClient
+    private nextId = 0
+
+    constructor(writable: Writable, readable: Readable) {
+        super()
+
+        this.client = new JSONRPCClient(
+            async request => {
+                const message = JSON.stringify(request)
+                const contentLength = Buffer.byteLength(message, 'utf-8')
+                writable.write(`Content-Length: ${contentLength}\r\n\r\n${message}`)
+            },
+            () => this.nextId++
+        )
+
+        let buffer = ''
+        readable.on('data', (chunk: Buffer) => {
+            buffer += chunk.toString()
+
+            while (true) {
+                const headerEnd = buffer.indexOf('\r\n\r\n')
```

Comment:
why is it \r\n? Is this test supposed to run in Windows only?

Reply:
It's a language server specification requirement: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#baseProtocol

Not required to be run on windows